pkg/controller/unifiedpushserver/backups.go

```diff
@@ -0,0 +1,130 @@
+package unifiedpushserver
+
+import (
+	"fmt"
+
+	pushv1alpha1 "github.com/aerogear/unifiedpush-operator/pkg/apis/push/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func backups(ups *pushv1alpha1.UnifiedPushServer) ([]batchv1beta1.CronJob, error) {
+	cronjobs := []batchv1beta1.CronJob{}
+	for _, upsBackup := range ups.Spec.Backups {
+		cronJobLabels := labels(ups, "backup")
+		for k, v := range upsBackup.Labels {
+			cronJobLabels[k] = v
+		}
+
+		jobLabels := map[string]string{"cronjob-name": upsBackup.Name}
+		for k, v := range cronJobLabels {
+			jobLabels[k] = v
+		}
+
+		cronjobs = append(cronjobs, batchv1beta1.CronJob{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      upsBackup.Name,
+				Namespace: ups.Namespace,
+				Labels:    cronJobLabels,
+			},
+			Spec: batchv1beta1.CronJobSpec{
+				Schedule: upsBackup.Schedule,
+				JobTemplate: batchv1beta1.JobTemplateSpec{
+					Spec: batchv1.JobSpec{
+						Template: corev1.PodTemplateSpec{
+							ObjectMeta: metav1.ObjectMeta{
+								Labels: jobLabels,
+							},
+							Spec: corev1.PodSpec{
+								// This SA needs to be created beforehand
+								// https://github.com/integr8ly/backup-container-image/tree/master/templates/openshift/rbac
+								ServiceAccountName: "backupjob",
+								Containers: []corev1.Container{
+									{
+										Name:            upsBackup.Name + "-keycloak-backup",
+										Image:           cfg.BackupImage,
+										ImagePullPolicy: "Always",
+										Command:         buildBackupContainerCommand(upsBackup, ups.Namespace),
+										Env:             buildBackupCronJobEnvVars(upsBackup, ups.Name, ups.Namespace),
+									},
+								},
+								RestartPolicy: corev1.RestartPolicyOnFailure,
+							},
+						},
+					},
+				},
+			},
+		})
+	}
+	return cronjobs, nil
+}
+
+func buildBackupContainerCommand(upsBackup pushv1alpha1.UnifiedPushServerBackup, upsNamespace string) []string {
+	command := []string{"/opt/intly/tools/entrypoint.sh", "-c", "postgres", "-n", upsNamespace}
+
+	// If there is no encryption secret, we need to inhibit the
+	// encryption behaviour
+	if upsBackup.EncryptionKeySecretName == "" {
```

Comment:
does the operator does with a situation where this var is filled?

Reply:
By default that `-e` option to the image gets a value of `gpg`: https://github.com/integr8ly/backup-container-image/blob/master/image/tools/entrypoint.sh#L7

That means that if we don't specify any other value, it will try to encrypt the backups. We have to explicitly set it to an empty string here to get it to not try encryption (the backup job would fail otherwise).

Reply:
I'm not sure what you're suggesting here. Is it to always put the `"-e", ""` in to the `command` slice on line 65? The problem with that is that it means encryption is off, no matter what you specify in the CR (I think the keycloak operator has this problem).

I guess we could have here:

```go
if upsBackup.EncryptionKeySecretName == "" {
	command = append(command, "-e", "")
} else {
	command = append(command, "-e", "gpg")
}
```

Unless you're talking about something completely different, which is entirely possible! :see_no_evil:

Reply:
Sorry I was talking about adding it in this line: https://github.com/aerogear/unifiedpush-operator/pull/32/files#diff-0ccc80f83580c98ef42e35739cdaaef1R65 intead of doing a `if` statement since we don't deal with gpg atm such as: `command := []string{"/opt/intly/tools/entrypoint.sh", "-c", "postgres",  "-e", "", "-n", upsNamespace}`

Reply:
Ah yeah, that's what I thought you meant. If we did that, we should also remove the encryption secret details from the CRD, since it would be misleading (like it is in the keycloak operator).

When the functionality is there in the backup-container-image project, I don't see a problem with having it possible to enable (which adding it unconditionally to the command string like that would mean).

With the way it is here, if someone has correctly specified an encryption secret, then it will work as expected and encrypt. Equally, if they don't specify one, it will work as expected, just not encrypt.

Reply:
ok makes sense :+1: